tests/testDoglegOptimizer.cpp

```diff
@@ -185,6 +183,120 @@ TEST(DoglegOptimizer, Constraint) {
 #endif
 }
 
+/* ************************************************************************* */
+TEST(DogLegOptimizer, VariableUpdate) {
```

Comment:
Please document why this example was created and credit the reporter/contributor as well in helping to craft this unit test.

Reply:
Done